src/http.rs

```diff
@@ -507,14 +549,24 @@ pub type RawHeaderLine = (String, Vec<u8>);
 /// >                ; obsolete line folding
 /// >                ; see Section 3.2.4
 /// > ```
-pub fn read_header<R: Reader>(stream: &mut R) -> HttpResult<Option<RawHeaderLine>> {
+pub fn read_header<R: Read>(stream: &mut R) -> HttpResult<Option<RawHeaderLine>> {
+    macro_rules! byte (
```

Comment:
Yea, it's gross. The change from an EOF error kind to Ok(0) makes this harder. I tried a ReadByte extension trait, but could propagate the correct error upon EOF. So, the macros allow inserting the appropriate HttpError when EOF is found early. 

I also have been working on a branch to use httparse, which removes much of this module.